packages/cli/src/commands/build/index.ts

```diff
@@ -774,3 +778,17 @@ function mergeWildcard(
   }
   return wildcard;
 }
+
+function mergeFlags(
+  buildResults: Iterable<BuildResult | BuildOutputConfig>
+): BuildResultV2Typical['flags'] {
+  let flags: Flag[] = [];
+
+  for (const result of buildResults) {
+    if ('flags' in result) {
+      flags.push(...(result.flags ?? []));
+    }
+  }
```

Comment:
I think `Array.prototype.flatMap` is more appropriate here. I find this code not very readable

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/flatMap